packages/admin/admin/src/form/FinalFormChooseFile.tsx

```diff
@@ -0,0 +1,269 @@
+        rejectedFileListItem: {
+            display: "flex",
+            padding: "8px 7px 8px 15px",
+            alignItems: "center",
+            gap: "10px",
+            borderRadius: "4px",
+            background: palette.background.default,
+            justifyContent: "space-between",
+            minWidth: "250px",
+            border: `1px dashed ${palette.error.main}`,
+            color: palette.error.main,
+            width: "100%",
+            boxSizing: "border-box",
+        },
+        errorMessage: {
+            display: "flex",
+            alignItems: "center",
+            color: palette.error.main,
+            gap: "5px",
+        },
+        droppableAreaIsDisabled: {},
+        droppableAreaHasError: {},
+        fileListText: {
+            textOverflow: "ellipsis",
+            overflow: "hidden",
+            whiteSpace: "nowrap",
+        },
+    });
+};
+
+export interface FinalFormChooseFileProps extends FieldRenderProps<File | File[], HTMLInputElement> {
+    disableDropzone?: boolean;
+    disableSelectFileButton?: boolean;
+    accept: Accept;
+    maxSize: number;
+    maxFiles?: number;
+    iconMapping?: {
+        delete?: React.ReactNode;
+        info?: React.ReactNode;
+        select?: React.ReactNode;
+    };
+}
+
+const FinalFormChooseFileComponent: React.FunctionComponent<WithStyles<typeof styles> & FinalFormChooseFileProps> = ({
+    classes,
+    disabled,
+    disableDropzone,
+    disableSelectFileButton,
+    accept,
+    maxSize = 50 * 1024 * 1024,
+    maxFiles,
+    input: { onChange, value: fieldValue, multiple: multipleFiles },
+    iconMapping = {},
+}) => {
+    const { delete: deleteIcon = <Delete />, info: infoIcon = <Info />, select: selectIcon = <Select /> } = iconMapping;
+
+    // For multipleFiles, if maxFiles is not set, or if maxFiles is set and the number of the files in the current fieldValue together with the acceptedFiles is equal or less then maxFiles, add acceptedFiles to the current fieldValue. Else replace the fieldValue with acceptedFiles
+    const onDrop = React.useCallback(
+        (acceptedFiles: File[]) => {
+            if (multipleFiles && Array.isArray(fieldValue)) {
+                if (!maxFiles || (maxFiles && fieldValue.length < maxFiles && fieldValue.length + acceptedFiles.length <= maxFiles)) {
+                    onChange([...fieldValue, ...acceptedFiles]);
+                } else onChange([...acceptedFiles]);
+            } else onChange([...acceptedFiles]);
+        },
+        [fieldValue, multipleFiles, onChange, maxFiles],
+    );
+
+    const removeFile = (removedFile: File) => () => {
+        const newFiles = Array.isArray(fieldValue) ? fieldValue.filter((file) => file !== removedFile) : undefined;
+        onChange(newFiles);
+    };
+
+    const { fileRejections, getRootProps, getInputProps, isDragReject } = useDropzone({
+        onDrop,
+        accept,
+        disabled: disabled || (maxFiles && fieldValue.length === maxFiles),
```

Comment:
Just for safety, maybe do `>=` incase the `maxFiles` is somehow exceeded 🤷🏼 

```suggestion
        disabled: disabled || (maxFiles && fieldValue.length >= maxFiles),
```